draft/XXXX-table-metadata/README.md

```diff
@@ -0,0 +1,80 @@
+---
+RFC: XXXX
+Author: Jacob Nesbitt
+Status: draft
+Created: 2020-10-29
+Last-Modified: 2020-10-29
+---
+
+# RFC XXXX: Table/Graph Metadata
+
+Currently, Multinet doesn't store any metadata about tables (or graphs), chiefly type information. This RFC outlines a way to specify, store and retrieve not just type information, but generalized metadata about a table/graph.
+
+## Motivation
+
+The main motivation for this is storing type information for each column in a table. We now have at least two use cases for storing this information.
+
+1. Basic computation/comparison. If we attempt to use these tables for creating graphs and deriving useful data, it's necessary that we store type information, so that we are able to properly compare and compute certain values. For example, if you have a column of numbers, but we store them as strings (the current behavior), then if you try to do comparisons of these column values between different entries, you either will be doing incorrect string comparisons, or will simply not be able to do the comparisons/computations you need to.
+
+2. Providing this information to other applications, so that they do not have to do this data inspection themselves, or if this information is required by those applications. Specifically, [UpSet](https://github.com/VCG/upset) is the first application for which this will be used.
+
+## Proposals
+
+#### Data Model
+There are several way we could go about this, but in my opinion, the cleanest way is to store this information in a special collection, in the workspace that the table resides in. This special collection would be named something like `multinet_meta`, and each entry would correspond to the metadata on a specific graph or table. I think the data model would look something like this:
+
+
+```
+{
+  item_id: str                 // The ID of the table/graph this data belongs to
+  item_type: "table" | "graph" // The major type of the item
+  table?: {                    // Only present if the major type is "table"
+    type: "edge" | "node"      // The table type
+    columns: [
+      {
+        key: str               // E.g. {"foo": "bar"}, "foo" is the key
+        type: "number" | "label" | "category" | "date" | "bool"
+      }
+    ]
+  }
+  graph?: {                    // Only present if the major type is "graph"
+    // TBD
+  }
+
+}
```

Comment:
Do we need to store the object type (table vs network) and then store metadata under differently named keys? Why not just dump `item_type` and then replace the `table`/`graph` entries by `metadata`?

Reply:
The main reason was to contain the likely different schemas under a possibly absent key, so that the definition doesn't contain a bunch of possibly absent keys at the top level.

I do think it's good to keep track of the type in each document (table vs graph), but I suppose it might not be strictly necessary, as the keys would imply this.

Reply:
There's a difference here between "required" and other metadata. What I've outlined above is "required" in the sense that we need that information in order to properly store the uploaded items, and provide the type information to external APIs in a structured way. I'd say if we want to add general metadata, that could go under a `"meta"` key or something, so that arbitrary metadata isn't mixed in with required metadata at the top level.

Reply:
Ok I think I understand. Let's go with your design then, since I'm having trouble explaining why I feel weird about this `table`/`graph` split. If that remains a problem in the future, we can always adjust if necessary.